### PR TITLE
breaking(minimal): handleBuild API for build

### PR DESCRIPTION
**Pull request by dai-shi**
also changes define-router

### CODE REVIEW

**Comment by tylersayshi** [comment]
this change is significant. I'm curious, what was the motivation for this refactor?

**Comment by dai-shi** [comment]
It's to support api routes. Long journey.

**Comment by tylersayshi** [comment]
> It's to support api routes. Long journey.

wow! I did not expect this to be so disruptive to the current code 😲

**Comment by dai-shi** [comment]
API routes are technically already possible with 39_api example.
But, I would like to build static files like RSS. It's completely a new capability.

File: examples/35_nesting/src/entries.tsx [hunk 1]
```diff
@@ -32,22 +33,43 @@ export default defineEntries({
...
+        })),
+        async () => ({
+          type: 'defaultHtml' as const,
+          pathname: '/',
+          htemlHead: generateHtmlHead(),
+        }),
+        async () => ({
+          type: 'defaultHtml' as const,
+          pathname: '/no-ssr',
+          htemlHead: generateHtmlHead(),
```
Comment (rmarscher) [inline examples/35_nesting/src/entries.tsx:70]: @dai-shi I think htemlHead might be a typo?

Comment (dai-shi) [reply examples/35_nesting/src/entries.tsx:70]: Oh, no. Yeah, thanks for pointing it out.

**Pull request merged**